### PR TITLE
 MS-50: Interviews - api rest

### PR DESCRIPTION
**Pull request by JamirMansilla**
**COMMITS**
- MS-50: add and set up the interviews.js file in models folder
 
- MS-50: edit functions in the controllers folder interviews.js file

- MS-50: add all the functions to the interviews.js file in controller´… 

- MS-50: add routes to the controllers

- MS-50: add the interviews.postman_collection.json file in collections… 

- MS-50: pritier interviews.js file in models folder

### CODE REVIEW

File: src/models/interviews.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+const InterviewsSchema = new Schema({
+  idCandidate: {
+    type: Schema.Types.ObjectId,
+    ref: "Candidates",
+    required: true,
+  },
+  idClient: {
+    type: Schema.Types.ObjectId,
+    ref: "Client",
```
Comment (nicolasblasco) [inline src/models/interviews.js:12]: @JamirMansilla we should change Client to Clients to follow the clients schema. @NicoMunoz909 changed it.

File: src/controllers/interviews.js [hunk 2]
```diff
@@ -1,66 +1,90 @@
...
   };
-  data.push(interview);
-  return res
-    .status(201)
-    .send(`Interview succesfully created! ${JSON.stringify(interview)}`);
+
+  if (req.body.dateTime) newInterview.dateTime = new Date(req.body.dateTime);
+
+  Interviews.create(newInterview);
+  res.status(201).json(newInterview);
```
Comment (nicolasblasco) [inline src/controllers/interviews.js:23]: @JamirMansilla we should add a callback or a promise to return an error response. We could use the same promise from getAll for this.

File: src/controllers/interviews.js [hunk 3]
```diff
@@ -1,66 +1,90 @@
...
 };
 
 const getAll = (req, res) => {
-  return res.status(200).json(data);
+  Interviews.find()
+    .then((interviews) => {
+      return res.status(200).json(interviews);
+    })
+    .catch((err) => {
+      return res.status(400).json(err);
```
Comment (nicolasblasco) [inline src/controllers/interviews.js:76]: @JamirMansilla we should change this HTTP response status code to 404: Not Found.

File: src/controllers/interviews.js [hunk 4]
```diff
@@ -1,66 +1,90 @@
...
-      .status(400)
-      .json({ Msg: "Could not find interview with specified ID" });
-  }
-  return res.status(200).json(data[index]);
+  Interviews.findById({ _id: new ObjectId(req.params.id) })
+    .then((interview) => {
+      return res.status(200).json(interview);
+    })
+    .catch((err) => {
+      return res.status(400).json(err);
```
Comment (nicolasblasco) [inline src/controllers/interviews.js:86]: @JamirMansilla we should change this HTTP response status code to 404: Not Found.

File: src/controllers/interviews.js (hunk 2 above)
Comment (JamirMansilla) [reply src/controllers/interviews.js:23]: already exist an error message

File: src/validations/interviews.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+const validate = (req, res, next) => {
+  const idCandidate = req.body.idCandidate;
+  const idClient = req.body.idClient;
+  const idPosition = req.body.idPosition;
+  const status = req.body.status;
+  if (idCandidate) {
```
Comment (damianalvarez) [inline src/validations/interviews.js:6]: @JamirMansilla I think that an improvement here is the ability to check against the DB whether the candidates, clients, and positions IDs exist or not. We can import the models and then make queries in order to check if they are valid or not.

**Review by damianalvarez — APPROVED** [review]

**Pull request merged**